src/factotum_server/server/tests.rs

```diff
@@ -0,0 +1,100 @@
+    server_manager.state = ::SERVER_STATE_DRAIN.to_string();
+    assert_eq!(server_manager.is_running(), false);
+}
+
+#[test]
+fn server_manager_get_start_time() {
+    let server_manager = ServerManager::new(Some("0.0.0.0".to_string()), 8080, "http://dummy.test/".to_string(), false);
+    assert_eq!(server_manager.get_start_time(), UTC::now().format("%F %T %Z").to_string());
+}
+
+#[test]
+fn server_manager_get_uptime() {
+    let server_manager = ServerManager::new(Some("0.0.0.0".to_string()), 8080, "http://dummy.test/".to_string(), false);
+    assert!(Regex::new(r"^\d+ Days, \d+ Hours, \d+ Minutes, \d+ Seconds$").unwrap().is_match(&server_manager.get_uptime()));
+}
+
+#[test]
+fn job_request_empty_job_name() {
+    let job_request = JobRequest::new("1", "", "/tmp/somewhere", vec![]);
+    let command_store = commands![::FACTOTUM.to_string() => "/tmp/fake_path".to_string()];
+    let validation_error = JobRequest::validate(job_request.clone(), &command_store).unwrap_err();
+    assert_eq!(validation_error, ValidationError::no_output("No valid value found: field 'jobName' cannot be empty".to_string()));
+}
+
+#[test]
+fn job_request_empty_factfile_path() {
+    let job_request = JobRequest::new("1", "dummy", "", vec![]);
+    let command_store = commands![::FACTOTUM.to_string() => "/tmp/fake_path".to_string()];
+    let validation_error = JobRequest::validate(job_request.clone(), &command_store).unwrap_err();
+    assert_eq!(validation_error, ValidationError::no_output("No valid value found: field 'factfilePath' cannot be empty".to_string()));
+}
+
+#[test]
+fn job_request_invalid_factfile_path() {
+    let job_request = JobRequest::new("1", "dummy", "/tmp/somewhere", vec![]);
+    let command_store = commands![::FACTOTUM.to_string() => "/tmp/fake_path".to_string()];
+    let validation_error = JobRequest::validate(job_request.clone(), &command_store).unwrap_err();
+    assert_eq!(validation_error, ValidationError::no_output("Value does not exist on host for 'factfilePath':'/tmp/somewhere'".to_string()));
+}
+
+#[test]
+fn job_request_can_append_job_args() {
+    let server_manager = ServerManager::new(Some("0.0.0.0".to_string()), 8080, "http://dummy.test/".to_string(), true);
+    let mut job_request = JobRequest::new("1", "dummy", "/tmp/somewhere", vec!["--first-arg".to_string()]);
+    JobRequest::append_job_args(&server_manager, &mut job_request);
+    assert_eq!(job_request.factfile_args, vec!["--first-arg", "--webhook", "http://dummy.test/", "--no-colour"]);
+}
+
+#[test]
+fn settings_request_is_valid() {
+    let settings_request = SettingsRequest::new(::SERVER_STATE_RUN);
+    let validated_settings_request = SettingsRequest::validate(settings_request.clone()).unwrap();
+    assert_eq!(validated_settings_request, settings_request);
+}
+
+#[test]
+fn settings_request_is_invalid() {
+    let settings_request = SettingsRequest::new("NOT A SERVER STATE");
```

Comment:
This test is for the function that validates the string input inside the JSON body. so I don't think it can work as an enum if checking string equality is required. Unless there's a way to do that with enums?